.github/workflows/nvqc_regression_tests.yml

```
@@ -0,0 +1,351 @@
+            if [[ -z "$1" ]]; then
+              publishing_run_ids=$(gh run -R NVIDIA/cuda-quantum list --workflow Publishing --json databaseId --jq .[].databaseId)
+            else
+              publishing_run_ids=$(gh run -R NVIDIA/cuda-quantum list --commit $1 --workflow Publishing --json databaseId --jq .[].databaseId)
+            fi
+            for run_id in $publishing_run_ids ; do
+                # Look into its jobs: if "CUDA Quantum Python wheels" matrix build was performed,
+                # then we have multiple jobs, like "CUDA Quantum Python wheels (python_arm64....")
+                cuda_wheel_build_jobs=$(gh run -R NVIDIA/cuda-quantum view $run_id --jq '.jobs.[] | select(.name | startswith("CUDA Quantum Python wheels (python_")).name' --json jobs)
+                if [ ! -z "$cuda_wheel_build_jobs" ]; then
+                  # This is a valid run that produces wheel artifacts
+                  echo $run_id
+                  break
+                fi
+            done
+          }
+
+          if [ -z "${workflow_id}" ]; then
+            workflow_id=$(get_publishing_run_id ${{ needs.metadata.outputs.cudaq_commit }})
+          fi
+          if [ ! -z "$workflow_id" ]; then
+            echo "Using artifacts from workflow id $workflow_id"
+            # Allow error when trying to download wheel artifacts since they might be expired.
+            set +e
+            gh run -R NVIDIA/cuda-quantum download $workflow_id --name "x86_64-py$python_version-wheels"
+            retVal=$?
+            set -e
+            if [ $retVal -ne 0 ]; then
+              echo "Failed to download wheels artifact from Publishing workflow run Id $workflow_id. Perhaps the artifacts have been expired."
+              # This is allowed since there might be a period where no Publishing workflow is run (e.g., no PR merged to main).
+              echo "skipped=true" >> $GITHUB_OUTPUT
+              exit 0
+            fi
+            python_version_filename=$(echo "${python_version//.}")
+            # Install Python and the wheel
+            apt-get update && apt-get install -y --no-install-recommends python$python_version python3-pip
+            wheelfile=$(find . -name "cuda_quantum*cp$python_version_filename*x86_64.whl")
+            python$python_version -m pip install $wheelfile
+            echo "skipped=false" >> $GITHUB_OUTPUT
+          else
+            echo "Failed to retrieve Publishing workflow run Id for commit ${{ needs.metadata.outputs.cudaq_commit }}"
+            exit 1
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Test NVQC
+        if: ${{ ! steps.install_wheel.skipped }}
+        run: |
+          echo "### Submit to NVQC from Python wheels" >> $GITHUB_STEP_SUMMARY
+          python_version=${{ inputs.python_version || env.python_version }}
+          export NVQC_API_KEY="${{ secrets.NVQC_SERVICE_KEY }}"
+          export NVQC_NCA_ID="${{ secrets.NVQC_NCA_ID }}"
```

I think this may need adjustment for this repo (as opposed to your repo). We should probably discuss this with @bettinaheim. If we are using the production NCA ID, then we don't need to specify the `NVQC_NCA_ID`.
```suggestion
          export NVQC_API_KEY="${{ secrets.NVQC_PROD_SERVICE_KEY }}"
```

---

Yes, having this workflow submit to the production environment is the intent, so Ben's change looks good. I am not sure what you mean by the "env variables for PROD"?